cgsmiles/graph_layout_utils.py

```diff
@@ -0,0 +1,368 @@
+    return points
+
+def check_and_fix_cis_trans(graph, points):
+    cis_trans = nx.get_node_attributes(graph, 'ez_isomer')
+    for  cis_trans_item in cis_trans.values():
+        for n1, n2, n3, n4, _type in cis_trans_item:
+            angle_init = _angle(n1, n2, n3, points)
+            angle_compl = _angle(n4, n3, n2, points)
+            if _type == 'trans' and np.isclose(angle_init,120, atol=10):
+                continue
+            if _type  == 'cis' and n1 < n4 and np.isclose(angle_init, 120, atol=10):
+                continue
+            if _type == 'trans':
+                angle = 120
+            elif n1 < n4:
+                angle = 120
+            else:
+                angle = 240
+
+            rotate_subgraph(graph,
+                            anchor=n2,
+                            reference=n3,
+                            target=n1,
+                            points=points,
+                            angle=angle)
+    return points
+
+def find_triplets(graph):
+    """
+    Find all triplets of nodes in the graph.
+
+    Parameters
+    ----------
+    graph: nx.Graph
+
+    Returns
+    -------
+    tuple([abc.hashable,abc.hashable,abc.hashable])
+        the node keys corresponding to the triplet
+        the central node is the common one.
+    """
+    triplets = set()
+    for edge1 in graph.edges:
+        for edge2 in graph.edges:
+            if edge1 != edge2:
+                common_node = set(edge1).intersection(edge2)
+                if common_node:
+                    common_node = common_node.pop()  # Extract the common node
+                    remaining_nodes = list(set(edge1 + edge2) - {common_node})
+                    triplet = (remaining_nodes[0], common_node, remaining_nodes[1])
+                    triplets.add(triplet)
+    return triplets
+
+
+def assign_angles(graph, default_angle=120):
+    """
+    Assign angle values for all angles in a molecule
+    graph. We assume they are all 120 if atoms have
+    2 or 3 neighbors. Otherwise, they are skipped. On
+    top we take care that five membered rings and triangles
+    have the correct angle sum.
+
+    Parameters
+    ----------
+    graph: nx.Graph
+
+    Returns
+    -------
+    tuple([[abc.hashable,abc.hashable,abc.hashable,int])
+        tuple with node keys corresponding to the angle
+        and the reference angle value
+    """
+    triplets = find_triplets(graph)
+    cycles = nx.cycle_basis(graph)
+    angles = []
+    for triplet in triplets:
+        # check if one is part of a ring
+        if any([triplet[0] in cycle for cycle in cycles]):
+            # find ring and check that all are part
+            for idx, cycle in enumerate(cycles):
+                if all([node in cycle for node in triplet]):
+                    angles.append((triplet[0],
+                                   triplet[1],
+                                   triplet[2],
+                                   180-360/len(cycle)))
```

Comment:
```suggestion
        # check if one is part of a ring
        rings = [cycle for cycle in cycles if set(triplet) <= set(cycle)]
        if rings:
            # find ring and check that all are part
            ring = rings[0]
            angles.append(triplet + (180-360/len(ring), ))
```

Reply:
This is for the inner angle of cycles. You can also say something about the *outer* angles. This happens when `len(set(triplet) & set(cycle)) == 2 and triplet[1] in cycle`, in which case the angle becomes `360/(2*len(ring))` (I think)

Reply:
Honestly, it doesn't matter for me. Things as they are work. Will reconsider it for #36